proxy/reverse.go

```diff
@@ -118,6 +121,9 @@ func (p *reverseProxy) ServeHTTP(rw http.ResponseWriter, clientreq *http.Request
 		redirectRequest(proxyreq, ep.URL)
 
 		res, err = p.transport.RoundTrip(proxyreq)
+		if requestCanceled {
```

Comment:
would you prefer an atomic operation or a channel for this? (I would go for sync/atomic)